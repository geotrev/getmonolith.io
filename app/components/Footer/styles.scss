@import "app/global";

.footer-wrapper {
  background: $color-dark-blue;

  .has-feather {
    @include create-link-color;
    display: inline-block;
    padding: $global-space;

    svg {
      @include create-flow-property("margin-left", 0);
    }
  }
}
