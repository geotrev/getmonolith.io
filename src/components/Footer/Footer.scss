@import "../../global";

.footer-wrapper {
  background: $brand-dark-blue;

  p {
    margin-bottom: 0;
  }

  .has-feather {
    display: inline-block;
    padding: 1rem;
    opacity: 0.8;

    &:hover,
    &:active,
    &:focus {
      opacity: 1;
    }
  }
}
