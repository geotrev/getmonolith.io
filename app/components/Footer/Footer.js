import React from "react"
import Github from "react-feather/dist/icons/github"
import Twitter from "react-feather/dist/icons/twitter"
import PropTypes from "prop-types"

import "./styles.scss"

export default function Footer(props) {
  const handleClick = event => {
    event.preventDefault()
    props.handleRefocusClick(props.headerRef)
  }

  return (
    <div className="footer-wrapper is-small-section is-fluid grid">
      <div className="row">
        <div className="has-no-padding column">
          <p className="has-no-margin-block-end has-text-center has-gray900-text">
            Undernet is a front-end framework created and maintained by{" "}
            <a className="has-white-text" href="http://www.geotrev.com">
              George Treviranus
            </a>
            .
          </p>
          <p className="has-no-margin-block-end has-text-center has-gray900-text">
            Animations provided by{" "}
            <a className="has-white-text" href="https://folgert.com/">
              Gavin Folgert
            </a>
            . ❤️
          </p>
          <ul className="has-text-center has-no-padding has-gray900-text has-unstyled-list has-direction-row has-display-flex has-justify-content-center">
            <li className="has-no-padding" role="none">
              <a
                role="menuitem"
                className="has-white-text has-feather"
                href="https://www.twitter.com/gwtrev"
              >
                <Twitter role="presentation" focusable="false" />
                <span className="is-visually-hidden">Open link to www.twitter.com/gwtrev</span>
              </a>
            </li>
            <li className="has-no-padding" role="none">
              <a
                role="menuitem"
                className="has-white-text has-feather"
                href="https://www.github.com/geotrev/undernet"
              >
                <Github role="presentation" focusable="false" />
                <span className="is-visually-hidden">
                  Open link to www.github.com/geotrev/undernet
                </span>
              </a>
            </li>
            <li className="has-no-padding" role="none">
              <button
                className="is-visually-hidden-focusable"
                onClick={handleClick}
                role="menuitem"
                type="button"
              >
                Return to top of page
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  )
}

Footer.propTypes = {
  handleRefocusClick: PropTypes.func.isRequired,
  headerRef: PropTypes.shape({
    current: PropTypes.instanceOf(Element),
  }),
}
