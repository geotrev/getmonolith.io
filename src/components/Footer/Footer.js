import React from "react"
import "./Footer.scss"

import Github from "react-feather/dist/icons/github"
import Twitter from "react-feather/dist/icons/twitter"

const Footer = () => {
  return (
    <div className="footer-wrapper small-section fluid grid">
      <div className="row">
        <div className="collapsed column">
          <p className="has-center-text">
            Undernet is front-end framework created and maintained by{" "}
            <a href="http://www.geotrev.com">George Treviranus</a>.
          </p>
          <p className="has-center-text">
            <a className="has-feather" href="https://www.twitter.com/gwtrev">
              <Twitter />
            </a>
            <a className="has-feather" href="https://www.github.com/geotrev/undernet">
              <Github />
            </a>
          </p>
        </div>
      </div>
    </div>
  )
}

export default Footer
