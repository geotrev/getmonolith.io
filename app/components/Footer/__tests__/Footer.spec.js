import React from "react"
import Footer from "../Footer"

jest.mock("react-feather/dist/icons/twitter", () => global.simpleMock("Twitter"))
jest.mock("react-feather/dist/icons/github", () => global.simpleMock("Github"))

describe("<Footer />", () => {
  const baseProps = {
    handleHeaderFocusClick: jest.fn(),
  }

  const mountComponent = () => {
    return mount(<Footer {...baseProps} />)
  }

  it("renders", () => {
    // Given
    const wrapper = mountComponent()
    // Then
    expect(wrapper).toMatchSnapshot()
  })

  it("calls handleHeaderFocusClick prop when hidden button is clicked", () => {
    // Given
    const wrapper = mountComponent()
    // When
    wrapper.find(".is-visually-hidden-focusable").simulate("click")
    // Then
    expect(baseProps.handleHeaderFocusClick).toHaveBeenCalled()
  })
})
