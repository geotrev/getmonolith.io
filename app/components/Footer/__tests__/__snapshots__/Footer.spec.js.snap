// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders 1`] = `
<Footer
  handleRefocusClick={[MockFunction]}
  headerRef={
    Object {
      "current": <header />,
    }
  }
>
  <div
    className="footer-wrapper is-small-section is-fluid grid"
  >
    <div
      className="row"
    >
      <div
        className="has-no-padding column"
      >
        <p
          className="has-no-margin-block-end has-text-center has-gray900-text"
        >
          Undernet is a front-end framework created and maintained by
           
          <a
            className="has-white-text"
            href="http://www.geotrev.com"
          >
            George Treviranus
          </a>
          .
        </p>
        <p
          className="has-no-margin-block-end has-text-center has-gray900-text"
        >
          Animations provided by
           
          <a
            className="has-white-text"
            href="https://folgert.com/"
          >
            Gavin Folgert
          </a>
          . ❤️
        </p>
        <ul
          className="has-text-center has-no-padding has-gray900-text has-unstyled-list has-direction-row has-display-flex has-justify-content-center"
        >
          <li
            className="has-no-padding"
            role="none"
          >
            <a
              className="has-white-text has-feather"
              href="https://www.twitter.com/gwtrev"
              role="menuitem"
            >
              <Twitter
                focusable="false"
                role="presentation"
              />
              <span
                className="is-visually-hidden"
              >
                Open link to www.twitter.com/gwtrev
              </span>
            </a>
          </li>
          <li
            className="has-no-padding"
            role="none"
          >
            <a
              className="has-white-text has-feather"
              href="https://www.github.com/geotrev/undernet"
              role="menuitem"
            >
              <Github
                focusable="false"
                role="presentation"
              />
              <span
                className="is-visually-hidden"
              >
                Open link to www.github.com/geotrev/undernet
              </span>
            </a>
          </li>
          <li
            className="has-no-padding"
            role="none"
          >
            <button
              className="is-visually-hidden-focusable"
              onClick={[Function]}
              role="menuitem"
              type="button"
            >
              Return to top of page
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</Footer>
`;
